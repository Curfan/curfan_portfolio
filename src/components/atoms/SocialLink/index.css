.social-link {
	display: flex;
	flex-shrink: 0;
	justify-content: center;
	align-items: center;
	width: 50px;
	height: 50px;
	background: #333b4e;
	border-radius: 50%;
	transition: background 0.3s ease-in;
	cursor: pointer;

	&:hover {
		background: white;
	}
}
