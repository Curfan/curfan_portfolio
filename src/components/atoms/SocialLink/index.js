import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const SocialLink = ({ className, Icon, link = '' }) => (
	<a
		className={classnames(styles.socialLink, className)}
		href={link}
		target="_blank"
		rel="noopener noreferrer"
	>
		{Icon && <Icon />}
	</a>
);

export default SocialLink;
