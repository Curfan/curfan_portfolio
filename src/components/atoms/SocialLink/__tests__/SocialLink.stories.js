import React from 'react';
import { storiesOf } from '@storybook/react';

import SocialLink from 'components/atoms/SocialLink';

import FacebookIcon from 'images/icon/social-facebook.inline.svg';

const stories = storiesOf('atoms/SocialLink', module);

stories.add('__interactive', () => (
	<SocialLink Icon={FacebookIcon} link="https://www.facebook.com/" />
));
