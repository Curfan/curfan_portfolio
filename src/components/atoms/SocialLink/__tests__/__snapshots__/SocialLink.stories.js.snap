// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <a
    class="socialLink"
    href="https://www.facebook.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <test-file-stub />
  </a>
</DocumentFragment>
`;
